Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/DirectPushService.cs

```diff
@@ -79,6 +79,8 @@ public async Task UpdateRegistration(IdentityAddress address, DeviceId deviceId,
 
             await _pnsRegistrationRepository.Update(registration, cancellationToken);
 
+            registration = await _pnsRegistrationRepository.FindByDeviceId(deviceId, cancellationToken, track: true);
```

Comment:
Why do you read it right after you updated it? You already have the up to date value.